functions/consumer/sftp-consumer/src/main/java/org/springframework/cloud/fn/consumer/sftp/SftpConsumerSessionFactoryConfiguration.java

```diff
@@ -18,27 +18,31 @@
 
 import org.apache.sshd.sftp.client.SftpClient;
 
-import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.context.annotation.Bean;
-import org.springframework.core.io.FileSystemResource;
-import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.core.io.Resource;
 import org.springframework.integration.file.remote.session.CachingSessionFactory;
 import org.springframework.integration.file.remote.session.SessionFactory;
 import org.springframework.integration.sftp.session.DefaultSftpSessionFactory;
+import org.springframework.lang.Nullable;
 
 /**
  * Session factory configuration.
  *
  * @author Gary Russell
  * @author Corneil du Plessis
  * @author Chris Bono
+ * @author Artem Bilan
  */
 public class SftpConsumerSessionFactoryConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public SessionFactory<SftpClient.DirEntry> sftpSessionFactory(SftpConsumerProperties properties, BeanFactory beanFactory) {
+	public SessionFactory<SftpClient.DirEntry> sftpSessionFactory(
+			SftpConsumerProperties properties,
+			@Value("#{${sftp.consumer.factory.known-hosts-expression:null}}") @Nullable Resource knownHosts) {
```

Comment:
This confused me for a second, as I was expecting the expression to be on the properties. However, it is in fact the path `sftp.consumer.factory.known-hosts-expression` to the field on the properties. I see your reason for doing this - let Spring inject it and we not have to write the following code :
```java
if (factory.getKnownHostsExpression() != null) {
  String location = factory.getKnownHostsExpression()
	  .getValue(IntegrationContextUtils.getEvaluationContext(beanFactory), String.class);
  sftpSessionFactory.setKnownHostsResource(resourceLoader.getResource(location));
}
```

Reply:
I think `@Resource` has its own limitations in Spring and technically it is more an equivalent to the `@Autowired`, not `@Value`.
And definitely it has nothing to do with the `org.springframework.core.io.Resource` 😄